libpod/events.go

```diff
@@ -38,6 +38,24 @@ func (c *Container) newContainerEvent(status events.Status) {
 		Attributes: c.Labels(),
 	}
 
+	if status == events.Create && c.runtime.config.Engine.EventsAuditContainerCreate {
+		err := func() error {
+			data, err := c.inspectLocked(true)
+			if err != nil {
+				return err
+			}
+			rawData, err := json.Marshal(data)
+			if err != nil {
+				return err
+			}
+			e.Attributes["inspect_data"] = string(rawData)
```

Comment:
I don't think we should use Attributes for that. Currently Attributes are set to the container label so if someone parses the Attributes they mistake this for a container label, and in case the `inspect_data` already exists you overwrite it.
It is also exposed via the compat API which could cause issues for API users.

Generally I wonder do you want this to only be logged or should podman events show it as well because this will make podman events basically unreadable (for a human) at the moment.

Reply:
> I don't think we should use Attributes for that. Currently Attributes are set to the container label so if someone parses the Attributes they mistake this for a container label, and in case the `inspect_data` already exists you overwrite it. It is also exposed via the compat API which could cause issues for API users.

Do you have an alternative in mind?

> Generally I wonder do you want this to only be logged or should podman events show it as well because this will make podman events basically unreadable (for a human) at the moment.

We could hide it by default in `podman-events` but allow users to query if needed.

Reply:
We could add a new field to the events.Details struct. Maybe call it `AuditData`?

Hiding the field from podman events by default makes most sense to me.

Reply:
> We could add a new field to the events.Details struct. Maybe call it `AuditData`?

I agree :+1:   Already started working on it.  I gravitated to call it `ContainerCreateInspectData` (also in containers.conf) but the name can be easily changed.